scipy/stats/tests/test_hypotests.py

```diff
@@ -90,97 +90,97 @@ def test_names(self):
-
-    def test_cdf_inf(self):
-        assert_allclose(
-                _cdf_cvm([0.02480, 0.03656, 0.11888, 1.16204]),
-                [0.01, 0.05, 0.5, 0.999],
-                atol=1e-4)
-
-    def test_cdf_support(self):
+    @pytest.mark.parametrize('n, x, ref', [
+        (4, [0.02983, 0.04111, 0.12331, 0.94251], [0.01, 0.05, 0.5, 0.999]),
+        (10, [0.02657, 0.03830, 0.12068, 0.56643], [0.01, 0.05, 0.5, 0.975]),
+        (1000, [0.02481, 0.03658, 0.11889, 1.16120], [0.01, 0.05, 0.5, 0.999]),
+        (None, [0.02480, 0.03656, 0.11888, 1.16204], [0.01, 0.05, 0.5, 0.999]),
+    ])
+    def test_cdf_ref(self, n, x, ref, xp):
+        xp_assert_close(_cdf_cvm(xp.asarray(x), n), xp.asarray(ref), atol=1e-4)
+
+    def test_cdf_support(self, xp):
         # cdf has support on [1/(12*n), n/3]
-        assert_equal(_cdf_cvm([1/(12*533), 533/3], 533), [0, 1])
-        assert_equal(_cdf_cvm([1/(12*(27 + 1)), (27 + 1)/3], 27), [0, 1])
+        xp_assert_equal(_cdf_cvm(xp.asarray([1/(12*533), 533/3]), 533),
+                        xp.asarray([0., 1.]))
+        xp_assert_equal(_cdf_cvm(xp.asarray([1/(12*(27 + 1)), (27 + 1)/3]), 27),
+                        xp.asarray([0., 1.]))
 
-    def test_cdf_large_n(self):
+    def test_cdf_large_n(self, xp):
         # test that asymptotic cdf and cdf for large samples are close
-        assert_allclose(
-                _cdf_cvm([0.02480, 0.03656, 0.11888, 1.16204, 100], 10000),
-                _cdf_cvm([0.02480, 0.03656, 0.11888, 1.16204, 100]),
-                atol=1e-4)
+        x = xp.asarray([0.02480, 0.03656, 0.11888, 1.16204, 100])
+        xp_assert_close(_cdf_cvm(x, 10000), _cdf_cvm(x), atol=1e-4)
 
-    def test_large_x(self):
+    def test_large_x(self, xp):
         # for large values of x and n, the series used to compute the cdf
         # converges slowly.
         # this leads to bug in R package goftest and MAPLE code that is
         # the basis of the implementation in scipy
         # note: cdf = 1 for x >= 1000/3 and n = 1000
-        assert_(0.99999 < _cdf_cvm(333.3, 1000) < 1.0)
-        assert_(0.99999 < _cdf_cvm(333.3) < 1.0)
+        x = xp.asarray(333.3, dtype=xp.float64)
+        assert (0.99999 < _cdf_cvm(x, 1000) < 1.0)
+        assert (0.99999 < _cdf_cvm(x) < 1.0)
 
-    def test_low_p(self):
+    def test_low_p(self, xp):
         # _cdf_cvm can return values larger than 1. In that case, we just
         # return a p-value of zero.
         n = 12
-        res = cramervonmises(np.ones(n)*0.8, 'norm')
-        assert_(_cdf_cvm(res.statistic, n) > 1.0)
-        assert_equal(res.pvalue, 0)
+        res = cramervonmises(xp.ones(n)*0.8, special.ndtr)
+        assert _cdf_cvm(res.statistic, n, xp=xp) > 1.0
+        xp_assert_equal(res.pvalue, xp.asarray(0.))
 
     @pytest.mark.parametrize('x', [(), [1.5]])
-    def test_invalid_input(self, x):
+    def test_invalid_input(self, x, xp):
         with pytest.warns(SmallSampleWarning, match=too_small_1d_not_omit):
```

Comment:
```suggestion
        with eager_warns(SmallSampleWarning, match=too_small_1d_not_omit, xp=xp):
```
maybe doesn't need skip?